src/darwin/Framework/CHIP/CHIPIMDispatch.mm

```diff
@@ -0,0 +1,284 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#import <Foundation/Foundation.h>
+
+#include <access/SubjectDescriptor.h>
+#include <app-common/zap-generated/att-storage.h>
+#include <app-common/zap-generated/callback.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app-common/zap-generated/ids/Commands.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandler.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/ConcreteCommandPath.h>
+#include <app/MessageDef/AttributeReportIBs.h>
+#include <app/MessageDef/StatusIB.h>
+#include <app/WriteHandler.h>
+#include <app/data-model/Decode.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/CHIPTLV.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/core/Optional.h>
+#include <protocols/interaction_model/Constants.h>
+
+/**
+ * This file defines the APIs needed to handle interaction model dispatch.
+ * These are the APIs normally defined in
+ * src/app/util/ember-compatibility-functions.cpp and the generated
+ * IMClusterCommandHandler.cpp but we want a different implementation of these
+ * to enable more dynamic behavior, since not all framework consumers will be
```

Comment:
With `CommandHandler`, `AttributeAccessInterface` and the ability to setup dynamic endpoints, is any of this really needed?

Reply:
For the specific use case here, this is less code than it takes to set up the relevant dynamic endpoints and the relevant other machinery as well as the access control provider, etc, etc.

Not least because we would need to fix existing issues around the fact that we can't have _only_ dynamic endpoints right now.

While we do want to fix those, I did not want to block this feature on that work.